### PR TITLE
Feature/op 586 edit therapy

### PR DESCRIPTION
**Pull request by tsognong**

### CODE REVIEW

File: src/components/accessories/dateField/DateField.tsx [hunk 1]
```diff
@@ -25,7 +26,13 @@ const DateField: FunctionComponent<IProps> = ({
...
 
   useEffect(() => {
-    setValue(fieldValue === "" ? null : new Date(+fieldValue));
+    if (fieldValue === "") {
+      setValue(null);
+    } else if (moment(+fieldValue).isValid()) {
+      setValue(moment(+fieldValue).toDate());
+    } else {
+      setValue(moment(fieldValue).toDate());
+    }
```
Comment (axelpezzo) [inline src/components/accessories/dateField/DateField.tsx:35]: See the comment on #165 MR

File: src/libraries/formatUtils/dataFormatting.ts [hunk 2]
```diff
@@ -14,9 +15,15 @@ export const opdDataFormatter = (
     diseases && data.disease3 !== ""
       ? diseases.find((el) => el.code === +data.disease3)
       : null;
```
Comment (axelpezzo) [inline src/libraries/formatUtils/dataFormatting.ts:17]: There is a reason why are you using a unary plus operator so frequently inside this function?

File: src/libraries/formatUtils/dataFormatting.ts [hunk 3]
```diff
@@ -14,9 +15,15 @@ export const opdDataFormatter = (
...
-
   return data;
 };
+
+export const renderDate = (date: string) => {
+  return moment(date).isValid()
+    ? moment(date).format("DD/MM/YYYY")
+    : moment(+date).isValid()
+    ? moment(+date).format("DD/MM/YYYY")
+    : "";
```
Comment (axelpezzo) [inline src/libraries/formatUtils/dataFormatting.ts:28]: I think this function could be write in this way:  

```
return moment(date).isValid()
    ? moment(date).format("DD/MM/YYYY")
    : "";
```

Do you agree?

File: src/libraries/reduxUtils/convert.ts [hunk 4]
```diff
@@ -7,7 +7,9 @@ export const convertToSummaryData = (
 ) => {
   const res = data.map(({ ...rest }) => ({
     type: field.type,
-    date: moment(rest[field.dateField]).format("DD/MM/YYYY"),
+    date: moment(rest[field.dateField]).isValid()
+      ? moment(rest[field.dateField]).format("DD/MM/YYYY")
+      : moment(+rest[field.dateField]).format("DD/MM/YYYY"),
```
Comment (axelpezzo) [inline src/libraries/reduxUtils/convert.ts:12]: Are you trying to format a date even if it's invalid?

File: src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx [hunk 5]
```diff
@@ -41,7 +42,10 @@ const TherapyForm: FC<TherapyProps> = ({
         name: "endDate",
         message: t("therapy.validatelastdate"),
         test: function (value) {
-          return moment(value).isSameOrAfter(moment(this.parent.startDate));
+          if (moment(+value).isValid()) {
+            return moment(+value).isSameOrAfter(moment(+this.parent.startDate));
+          } else
+            return moment(value).isSameOrAfter(moment(this.parent.startDate));
```
Comment (axelpezzo) [inline src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx:48]: As below, if the date is invalid it shouldn't be formatted.

File: src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import { Checkbox, FormControlLabel } from "@material-ui/core";
+import id from "date-fns/esm/locale/id/index.js";
```
Comment (axelpezzo) [inline src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx:2]: Is this an import mistake or are you gonna use in somewhere?

File: src/components/accessories/patientTherapy/PatientTherapy.tsx [hunk 7]
```diff
@@ -4,123 +4,139 @@ import TherapyForm from "./therapyForm/TherapyForm";
...
+      {(createStatus === "FAIL" || updateStatus === "FAIL") && (
+        <div ref={infoBoxRef} className="info-box-container">
+          <InfoBox type="error" message={t("common.somethingwrong")} />
+        </div>
+      )}
+
       <ConfirmationDialog
-        isOpen={hasSucceeded}
-        title="Therapy Created"
+        isOpen={createStatus === "SUCCESS" || updateStatus === "SUCCESS"}
```
Comment (axelpezzo) [inline src/components/accessories/patientTherapy/PatientTherapy.tsx:123]: I would handler this behaviour with a useState: use an useEffect in order to properly update the state once one of this variables will change, do you agree?

Comment (axelpezzo) [inline src/components/accessories/patientTherapy/PatientTherapy.tsx:123]: I would handler this behaviour with an useState: use an useEffect in order to properly update the state once one of this variables will change, do you agree?

File: src/libraries/formatUtils/dataFormatting.ts (hunk 3 above)
Comment (tsognong) [reply src/libraries/formatUtils/dataFormatting.ts:28]: To handle timestamp formatting we should use moment(+date).format("format string). So if we want to ignore date string we should use 
`
return moment(+date).isValid()
    ? moment(+date).format("DD/MM/YYYY")
    : "";
`

File: src/libraries/reduxUtils/convert.ts (hunk 4 above)
Comment (tsognong) [reply src/libraries/reduxUtils/convert.ts:12]: We have two cases: millisecondes and date string. invalid date is also possible.

File: src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx (hunk 6 above)
Comment (tsognong) [reply src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx:2]: mistake

File: src/components/accessories/patientTherapy/PatientTherapy.tsx (hunk 7 above)
Comment (tsognong) [reply src/components/accessories/patientTherapy/PatientTherapy.tsx:123]: May be we can do that. But the useSelector is a hook. Is it not duplicating the logic?

File: src/libraries/formatUtils/dataFormatting.ts (hunk 2 above)
Comment (tsognong) [reply src/libraries/formatUtils/dataFormatting.ts:17]: We have to map each **disease code** gives by the form to the correspondant object.

File: src/libraries/formatUtils/dataFormatting.ts (hunk 3 above)
Comment (tsognong) [reply src/libraries/formatUtils/dataFormatting.ts:28]: May be we should use return moment(+date).isValid()
    ? moment(+date).format("DD/MM/YYYY")
    : ""; As we are working with timestamp

File: src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx (hunk 5 above)
Comment (tsognong) [reply src/components/accessories/patientTherapy/therapyForm/TherapyForm.tsx:48]: OK

File: src/libraries/formatUtils/dataFormatting.ts (hunk 2 above)
Comment (tsognong) [reply src/libraries/formatUtils/dataFormatting.ts:16]: The disease returned by formik is just a string code, we have to convert it to the entire object in order to call the api. That is why I am using the + operator to convert that string to integer in order to find the right object.

File: src/libraries/reduxUtils/convert.ts (hunk 4 above)
Comment (tsognong) [reply src/libraries/reduxUtils/convert.ts]: OK, I changed it

File: src/libraries/formatUtils/dataFormatting.ts (hunk 3 above)
Comment (axelpezzo) [reply src/libraries/formatUtils/dataFormatting.ts]: Is this already done?

**Review by axelpezzo — APPROVED** [review]

File: src/libraries/formatUtils/dataFormatting.ts (hunk 3 above)
Comment (tsognong) [reply src/libraries/formatUtils/dataFormatting.ts]: OK, I applied since.

**Pull request merged**